src/main/java/uk/gov/hmcts/sptribs/common/event/CreateTestCase.java

```diff
@@ -129,23 +136,73 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<C
     public SubmittedCallbackResponse submitted(CaseDetails<CaseData, State> details,
                                                CaseDetails<CaseData, State> beforeDetails) {
         var data = details.getData();
-        String claimNumber = data.getHyphenatedCaseRef();
 
-        sendApplicationReceivedNotification(claimNumber, data);
+        setSupplementaryData(details.getId());
+
+        String claimNumber = data.getHyphenatedCaseRef();
+        try {
+            sendApplicationReceivedNotification(claimNumber, data);
+        } catch (Exception notificationException) {
+            log.error("Create case notification failed with exception : {}", notificationException.getMessage());
+            return SubmittedCallbackResponse.builder()
+                .confirmationHeader(format("# Create case notification failed %n## Please resend the notification"))
+                .build();
+        }
 
         return SubmittedCallbackResponse.builder()
             .confirmationHeader(format("# Case Created %n## Case reference number: %n## %s", claimNumber))
             .build();
     }
 
+    private void initialiseFlags(CaseData data) {
+        data.setCaseFlags(Flags.builder()
+            .details(new ArrayList<>())
+            .partyName(null)
+            .roleOnCase(null)
+            .build());
+
+        data.setSubjectFlags(Flags.builder()
+            .details(new ArrayList<>())
+            .partyName(data.getCicCase().getFullName())
```

Comment:
It will be safe to put a null check on full name (just in case we change it to optional in the future)

Reply:
Added null check.